filters/imagefilter.go

```diff
@@ -48,25 +49,12 @@ func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
 
 	rsp.Header.Del("Content-Length")
 
-	in := rsp.Body
-	r, w := io.Pipe()
-	rsp.Body = r
-
-	go handleImageTransform(w, in, f)
+	handleImageTransform(rsp, f)
 }
 
-func handleImageTransform(out *io.PipeWriter, in io.ReadCloser, f ImageFilter) error {
+func handleImageTransform(rsp *http.Response, f ImageFilter) error {
 	var err error
-
-	defer func() {
```

Comment:
see my other comments. I think it can be simplified by trying to keep the rsp object out of the image business. The image processing functions only need the body.